helm/myapp-chart/templates/istio/virtualservices.yml

```diff
@@ -0,0 +1,80 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: {{ .Release.Name }}-frontend-vs
+spec:
+  hosts:
+  - "{{ .Values.ingress.host }}"
+  gateways:
+  - {{ .Release.Name }}-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: {{ .Release.Name }}-app-frontend-service
+        subset: v1
+        port:
+          number: {{ .Values.appFrontend.servicePort | default 80 }}
+      weight: 90
+    - destination:
+        host: {{ .Release.Name }}-app-frontend-service
+        subset: v2
+        port:
+          number: {{ .Values.appFrontend.servicePort | default 80 }}
+      weight: 10
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: {{ .Release.Name }}-app-service-vs
+spec:
+  hosts:
+  - "{{ .Values.ingress.host }}"
+  gateways:
+  - {{ .Release.Name }}-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /app/
+    route:
+    - destination:
+        host: {{ .Release.Name }}-app-service
+        subset: v1
+        port:
+          number: {{ .Values.appService.servicePort | default 80 }}
+      weight: 90
+    - destination:
+        host: {{ .Release.Name }}-app-service
+        subset: v2
+        port:
+          number: {{ .Values.appService.servicePort | default 80 }}
+      weight: 10
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: {{ .Release.Name }}-model-service-vs
+spec:
+  hosts:
+  - "{{ .Values.ingress.host }}"
+  gateways:
+  - {{ .Release.Name }}-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /model/
+    route:
+    - destination:
+        host: {{ .Release.Name }}-model-service
+        subset: v1
+        port:
+          number: {{ .Values.modelService.servicePort | default 80 }}
+      weight: 90
+    - destination:
+        host: {{ .Release.Name }}-model-service
+        subset: v2
+        port:
+          number: {{ .Values.modelService.servicePort | default 80 }}
+      weight: 10 
```

Comment:
Here too... don't these virtual services require versioned deployments?

Reply:
Update: Since we haven't added any versioned deployments here we shouldn't add any subsets.